JitsiConference.js

```diff
@@ -1263,12 +1269,22 @@ JitsiConference.prototype.removeTrack = function(track) {
  */
 JitsiConference.prototype.replaceTrack = function(oldTrack, newTrack) {
     const oldVideoType = oldTrack?.getVideoType();
+    const mediaType = oldTrack?.getType() || newTrack?.getType();
     const newVideoType = newTrack?.getVideoType();
 
     if (FeatureFlags.isMultiStreamSupportEnabled() && oldTrack && newTrack && oldVideoType !== newVideoType) {
         throw new Error(`Replacing a track of videoType=${oldVideoType} with a track of videoType=${newVideoType} is`
             + ' not supported in this mode.');
     }
+
+    if (FeatureFlags.isSourceNameSignalingEnabled() && newTrack) {
```

Comment:
In case of a real transfer, the local SSRC gets updated here https://github.com/jitsi/lib-jitsi-meet/blob/master/modules/RTC/TraceablePeerConnection.js#L2009 since it doesn't trigger a O/A cycle. For other cases where a null track is replaced by a new track, the source name needs to be set before the O/A cycle so that the SSRCs are extracted from the SDP and cached.

Reply:
hmm I don't understand... you're writing about SSRCs and I'm writing about source names?

Reply:
The new track can still inherit it here, otherwise its being set here - https://github.com/jitsi/lib-jitsi-meet/blob/f772b108d40982aecacbce884f593c24de69baf2/JitsiConference.js#L1367

Reply:
The reason the source names are being set before the track replace is successful as opposed to after is that the we need the source name for processing the local SDP and produce the TPC localSSRCs cache. A real replace operation doesn't change the SSRCs and therefore it doesn't matter there whether we set it before or after the track replace is successful.

Reply:
hm if it's done in setup new track already then it seems like there's too many places where it's being set, because once it will be set here and then again on the line 1309... what if there's a bug later one the source name will start to change? I think it will be hard to detect such bugs if we have many places where it's set

Reply:
is not set source name in setup track and do it only in addTrack and replaceTrack instead at the timing when we need it?

Reply:
It will be set in line 1309 only it hasn't been set already.

Reply:
Also, addTrack and replaceTrack don't cover the case when the tracks are added to the conference before session creation. Therefore we need to source name after the tracks are added to the pc in that case.

Reply:
👍